crates/database/src/events.rs

```diff
@@ -38,14 +38,17 @@ pub struct PreSignature {
 #[derive(Clone, Copy, Debug, Default)]
 pub struct EventIndex {
     pub block_number: i64,
-    pub block_hash: BlockHash,
     pub log_index: i64,
 }
 
-pub async fn last_block(ex: &mut PgConnection) -> Result<(i64, BlockHash), sqlx::Error> {
-    const QUERY: &str =
-        "SELECT block_number, block_hash FROM blocks ORDER_BY block_number DESC LIMIT 1;";
-    sqlx::query_as(QUERY).fetch_one(ex).await
+pub async fn last_block(ex: &mut PgConnection) -> Result<i64, sqlx::Error> {
+    const QUERY: &str = "\
+            SELECT GREATEST( \
+                (SELECT COALESCE(MAX(block_number), 0) FROM trades), \
+                (SELECT COALESCE(MAX(block_number), 0) FROM settlements), \
+                (SELECT COALESCE(MAX(block_number), 0) FROM invalidations), \
+                (SELECT COALESCE(MAX(block_number), 0) FROM presignature_events));";
```

Comment:
In the meantime some [code](https://github.com/cowprotocol/services/pull/499/files) for the ETH flow got merged which added the table `onchain_placed_orders` that also has a block number.